### ISSUE TITLE
7B INT8 ConvRot model fails to load on Windows ROCm ("Only Tensors of floating point and complex dtype can require gradients")

### ISSUE THREAD

**Issue by iamXinc** [patientx-cfz/comfyui-rocm#92]
Hello patientx,

Need your help again. Here are the issue and system details (drafted with the help from ChatGPT):

Environment
OS - Windows
ComfyUI - ComfyUI 0.27.0 (Revision 089971b8, Released 2026-07-08)
ComfyUI Fork - PatientX portable ROCm build (comfyui-rocm-061726-multiarch)
Python - Python 3.12.9
PyTorch - 2.12.0+rocm7.15.0a20260708
ROCm - ROCm 7.15
GPU -  AMD Radeon 8050S Graphics (gfx1151), VRAM: 22,393 MB, RAM: 16,024 MB
Memory - Dynamic VRAM enabled, Smart Memory disabled, Async weight offloading enabled, Sage Attention enabled
SeedVR2 - ComfyUI-SeedVR2_VideoUpscaler v2.5.24 (latest from ComfyUI Manager)
Models - DiT: seedvr2_7b_int8_convrot.safetensors (official Comfy-Org HuggingFace release)
VAE: ema_vae_fp16.safetensors

Issue - The official 7B INT8 ConvRot model fails during model loading before inference starts.
The node successfully:
creates the DiT model
creates the VAE
starts generation
encodes the image
begins materializing the DiT weights
It then immediately fails while loading the state_dict.

Error: Error(s) in loading state_dict for NaDiT: Only Tensors of floating point and complex dtype can require gradients
This repeats for many INT8 weight tensors such as:
blocks.0.attn.proj_qkv.vid.weight
blocks.0.attn.proj_qkv.txt.weight
blocks.0.attn.proj_out.vid.weight
...
The error occurs before inference begins.

What works - Official 3B FP8 model works correctly. Other INT8 ConvRot models (e.g. Krea2/SCAIL2 INT8) work correctly in the same ComfyUI installation. This appears to be specific to loading the SeedVR2 7B INT8 ConvRot checkpoint.

Additional observations

The startup log reports that comfy_kitchen exposes the required ConvRot quantization/dequantization capabilities, including:
quantize_int8_convrot_weight
dequantize_int8_convrot_weight
which suggests the runtime has ConvRot support available.

I have verified:
Latest SeedVR2 node (v2.5.24)
Official Comfy-Org HuggingFace model
Latest PatientX ROCm portable build
Workflow is otherwise unchanged except for replacing the DiT model.

Is there an additional requirement for the 7B INT8 ConvRot model, or could this be a compatibility issue in the SeedVR2 loader on Windows ROCm?

Thank you

**Comment by Jules-A** [comment]
Are you using the custom INT8 node? https://github.com/patientx/ComfyUI-INT8-Fast-ROCM
Apparently the official support isn't great and the custom node runs faster anyway but I haven't tested the official implementation in weeks. I'd try that node first anyway and see if that fixes it first.

**Comment by patientx** [comment]
> Thank you

First make sure comfyui itself is updated to a version which supports int8 in the first place, then use my int8-fast-rocm loader 
If you want to update comfyui-rocm and rocm-pytorch packages to the latest ones available run these batch files in order : 
(`comfyui-rocm-updater.bat`) & (`rocm-pytorch-package-updater.bat`)

<img width="1596" height="669" alt="Image" src="https://github.com/user-attachments/assets/6daf69be-07a2-40bf-8403-a8760d6d0aab" />

it works.

**Comment by iamXinc** [comment]
Thank you, my INT8 node does work as expected on Krea2 and SCAIL2. It's just that my SeedVR2 workflow only had LoadDIT model and no option to connect/add INT8 Diffusion model. Here's a pic below, I am trying to recreate the workflow pic you shared and can't find those pre-process, apply conditioning and post process nodes. Could you please share the workflow with me or provide a link to it. I checked SeedVR2 git page as well but they don't have the INT8 workflows there either. Many thanks :).

<img width="2052" height="1000" alt="Image" src="https://github.com/user-attachments/assets/0bea86d9-57d4-4793-938b-eb51cf0a18a4" />

**Comment by patientx** [comment]
you are using the custom node, comfy has native seedvr2 support now. if you can't find, just search templates for seedvr2 int8 image upscale for example and change the diffusion loader to int8-fast-rocm loader in there...

**Comment by iamXinc** [comment]
@patientx  - That was it, thank you. I updated to the latest build and got the missing seedvr2 conditioning and post-processing nodes. I did have to roll back to the previous build due to issues with LTX nodes and another frontend bug. Thanks again.

**Comment by iamXinc** [comment]
> Are you using the custom INT8 node? https://github.com/patientx/ComfyUI-INT8-Fast-ROCM Apparently the official support isn't great and the custom node runs faster anyway but I haven't tested the official implementation in weeks. I'd try that node first anyway and see if that fixes it first.

Thanks, I did have that node but needed to update to the latest build for some Comfy built in SeedVR2 nodes.
